lib/venue.rb

```diff
@@ -1,2 +1,38 @@
 class Venue
+  attr_accessor :name, :capacity, :patrons
+
+  def initialize(name, capacity)
+    @name = name
+    @capacity = capacity
+    @patrons = []
+  end
+
+  def add_patron(patron_name)
+    # Add addition patrons to the venue list.
+    @patrons << patron_name
+  end
+
+  def yell_at_patrons
+    # Iterate through patrons and change string value to upcase.
+    @patrons.map do |patron|
+      patron.upcase
+    end
+  end
+
+  def over_capacity?
+    # If the number of patrons is greater than capacity return true.
+    @patrons.count > @capacity ? true : false
+  end
+
+  def kick_out
+    if self.over_capacity? == true
```

Comment:
Can you think of a way to refactor this method using a `while` or `until` loop?

Reply:
I think this would do. It worked in my current test. Way cleaner. 

``` Ruby   
while over_capacity? == true do
      @patrons.pop
end
```